pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -671,9 +671,58 @@ func getServerByName(client *gophercloud.ServiceClient, name types.NodeName) (*S
 	return &serverList[0], nil
 }
 
-func nodeAddresses(srv *servers.Server, networkingOpts NetworkingOpts) ([]v1.NodeAddress, error) {
+// IP addresses order:
+// * interfaces private IPs
+// * access IPs
+// * metadata hostname
+// * server object Addresses
+func nodeAddresses(srv *servers.Server, interfaces []attachinterfaces.Interface, networkingOpts NetworkingOpts) ([]v1.NodeAddress, error) {
 	addrs := []v1.NodeAddress{}
 
+	// parse private IP addresses first in an ordered manner
+	for _, iface := range interfaces {
+		for _, fixedIP := range iface.FixedIPs {
```

Comment:
I think we only care about the nic with property `"port_state": "ACTIVE"`

Reply:
added an extra check